antlr/src/main/antlr/FSH.g4

```diff
@@ -96,7 +96,7 @@ value:              SEQUENCE | STRING | MULTILINE_STRING | NUMBER | DATETIME | T
 item:               SEQUENCE (KW_NAMED SEQUENCE)? CARD flag*;
 code:               CODE STRING?;
 concept:            STAR code (STRING | MULTILINE_STRING)?;
-quantity:           NUMBER UNIT STRING?;
+quantity:           NUMBER (UNIT | CODE)? STRING?;
```

Comment:
I think `(UNIT | CODE)?` should be `(UNIT | CODE)`, no `?`, since we want one or the other. And we assume that one or the other is present in later processing.

Reply:
Hmmm... on one hand, it might be nice to support `1.0 "hot fudge sundae"` for a quantity that uses only the human-readable unit and no code.

On the other hand, since everything but `NUMBER` is optional in the grammar above, now `1.0` alone would match on `quantity` -- which isn't really our desired behavior.

Since the prior behavior required `UNIT` and this PR is just about also allowing `CODE`, I think I agree w/ Nick -- it should not be optional.  If we want to make it optional, that should be a different task.

Reply:
Understood! This makes a lot fo sense, I'll change this so that either `UNIT` or `CODE` is required.